### PR TITLE
grafana datasource v1

### LINKED ISSUE #7011: add grafana datasource plugin v1
- customer POC requirements
    - graph of top api latencies over time
    - graph of number of request errors over time, grouped by endpoint
        - distribution of HTTP status codes over time

Plugin spec

- drop down of the product you want to query, starting w `traces`
- text input for a query. can use the same syntax as on [app.highlight.io](http://app.highlight.io/)?
- picker for fields to query
- picker for filters
- picker for group by

### PR DESCRIPTION
**Pull request by mayberryzane**
## Summary
- adds a grafana data source for Highlight which uses the `traces_metrics` gql resolver
- allows the user to select the table, metric, column, group by, and query
- closes #7011
- https://www.loom.com/share/4c14dc5c38514312a26048c0027eeff5

## How did you test this change?
- click tested locally

## Are there any deployment considerations?
- no

## Does this work require review from our design team?
- no

### CODE REVIEW

File: sdk/highlight-grafana-datasource/src/datasource.ts [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                },
+              },
+              column: target.column,
+            },
+            query: GET_TRACES_METRICS,
+          }),
+          {}
+        );
+        if (response.errors?.length) {
+          throw response.errors.map((e) => JSON.stringify(e)).join(', ');
```
Comment (Vadman97) [inline sdk/highlight-grafana-datasource/src/datasource.ts:117]: should we pull this out into some function that is driven by the query type? if we have the dropdown for something other than traces, how would we choose what query we make here?

File: sdk/highlight-grafana-datasource/src/components/QueryEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    { value: 'ServiceVersion', label: 'service_version' },
+    { value: 'http.status_code', label: 'http.status_code' },
+  ];
+
+  return (
+    <div className="gf-form">
+      <Field label="Table">
+        <Select value={table} onChange={onTableChange} options={tableOptions} />
+      </Field>
+      <Field label="Metric" error="why isn't this displayed???">
```
Comment (Vadman97) [inline sdk/highlight-grafana-datasource/src/components/QueryEditor.tsx:65]: `error` value from testing?

File: sdk/highlight-grafana-datasource/src/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+<!-- This README file is going to be the one displayed on the Grafana.com website for your plugin. Uncomment and replace the content here before publishing.
```
Comment (Vadman97) [inline sdk/highlight-grafana-datasource/src/README.md:1]: could we have a description of highlight / the plugin for publishing in the future?

File: package.json
Comment (Vadman97) [inline package.json:1]: could the `test:all` script run the grafana datasource tests?

**Review by Vadman97 — APPROVED** [review]

File: sdk/highlight-grafana-datasource/src/datasource.ts (hunk 1 above)
Comment (mayberryzane) [reply sdk/highlight-grafana-datasource/src/datasource.ts:117]: yeah I think having a mapping from `target.table` to the function returning a data frame is probably the way to go, can do this when we add another table to the dropdown

File: sdk/highlight-grafana-datasource/src/README.md (hunk 3 above)
Comment (mayberryzane) [reply sdk/highlight-grafana-datasource/src/README.md:1]: 👍 will get a PR out tomorrow with this + grafana datasource tests

**Pull request merged**